### PR TITLE
FOGL-7839: Improvements in valgrind test to check memory leakage

### PR DESCRIPTION
**Pull request by Mohit04tomar**

### CODE REVIEW

File: tests/system/memory_leak/scripts/setup [hunk 1]
```diff
@@ -40,17 +41,29 @@ valgrind_conf=' --tool=memcheck --leak-check=full --show-leak-kinds=all'
...
+fi
 
 pnorth_C=${FLEDGE_ROOT}/scripts/services/north_C
 echo $pnorth_C
 sudo sed -i 's#/usr/local/fledge#'"$FLEDGE_ROOT"'#g' ${pnorth_C}
-sudo sed -i '/.\/fledge.services.north.*/s/^/valgrind --log-file=\/tmp\/north_valgrind.log '"$valgrind_conf"' /' ${pnorth_C}
+if [[ "${COLLECT_FILES}" == "LOGS" ]]; then
+	sudo sed -i '/.\/fledge.services.north.*/s/^/valgrind --log-file=\/tmp\/north_valgrind.log '"$valgrind_conf"' /' ${pnorth_C}
+else
+	sudo sed -i '/.\/fledge.services.south.*/s/^/valgrind --xml=yes --xml-file=\/tmp\/north_valgrind_%p.xml --track-origins=yes '"$valgrind_conf"' /' ${psouth_c}
```
Comment (ashish-jabble) [inline tests/system/memory_leak/scripts/setup:56]: it should be with **north** Not south

File: tests/system/memory_leak/scripts/setup [hunk 2]
```diff
@@ -40,17 +41,29 @@ valgrind_conf=' --tool=memcheck --leak-check=full --show-leak-kinds=all'
...
+fi
 
 pstorage=${FLEDGE_ROOT}/scripts/services/storage
 echo $pstorage
 sudo sed -i 's#/usr/local/fledge#'"$FLEDGE_ROOT"'#g' ${pstorage}
-sudo sed -i '/\${storageExec} \"\$@\"/s/^/valgrind --log-file=\/tmp\/storage_valgrind.log '"$valgrind_conf"' /' ${pstorage}
+if [[ "${COLLECT_FILES}" == "LOGS" ]]; then
+	sudo sed -i '/\${storageExec} \"\$@\"/s/^/valgrind --log-file=\/tmp\/storage_valgrind.log '"$valgrind_conf"' /' ${pstorage}
+else
+	sudo sed -i '/.\/fledge.services.south.*/s/^/valgrind --xml=yes --xml-file=\/tmp\/storage_valgrind_%p.xml --track-origins=yes '"$valgrind_conf"' /' ${psouth_c}
```
Comment (ashish-jabble) [inline tests/system/memory_leak/scripts/setup:65]: it should be with **storage** Not south

File: tests/system/memory_leak/test_memcheck.sh [hunk 3]
```diff
@@ -9,6 +9,8 @@ export FLEDGE_ROOT=$(pwd)/fledge
 
 FLEDGE_TEST_BRANCH="$1"    # here fledge_test_branch means branch of fledge repository that is needed to be scanned, default is develop
 
+COLLECT_FILES="$2"
```
Comment (ashish-jabble) [inline tests/system/memory_leak/test_memcheck.sh:12]: It should have some default value; And if pass then accept that. Also we may add validation to accept possible values like  XML, LOGS

File: tests/system/memory_leak/test_memcheck.sh [hunk 4]
```diff
@@ -116,7 +118,11 @@ generate_valgrind_logs(){
   echo 'Creating reports directory';
   mkdir -p reports/test1 ; ls -lrth
   echo 'copying reports '
-  cp -rf /tmp/*valgrind*.log /tmp/*valgrind*.xml reports/test1/. && echo 'copied'
+  if [[ "${COLLECT_FILES}" == "LOGS" ]]; then
+   cp -rf /tmp/*valgrind*.log reports/test1/. && echo 'copied'
+  else
+   cp -rf /tmp/*valgrind*.xml reports/test1/. && echo 'copied'
+  fi
```
Comment (ashish-jabble) [inline tests/system/memory_leak/test_memcheck.sh:125]: More simpler way
```
extension="xml"
if [[ "${COLLECT_FILES}" == "LOGS" ]]; then extension="log"; fi
cp -rf /tmp/*valgrind*.${extension} reports/test1/. && echo 'copied'
```

File: tests/system/memory_leak/test_memcheck.sh [hunk 5]
```diff
@@ -116,7 +118,11 @@ generate_valgrind_logs(){
   echo 'Creating reports directory';
   mkdir -p reports/test1 ; ls -lrth
```
Comment (ashish-jabble) [inline tests/system/memory_leak/test_memcheck.sh:119]: Not sure why are we creating reports directory with **test1**? What does it really mean?

**Review by ashish-jabble — CHANGES_REQUESTED** [review]

File: tests/system/memory_leak/scripts/setup [hunk 6]
```diff
@@ -3,6 +3,7 @@
 set -e
 
 BRANCH=${2:-develop}   # here Branch means branch of fledge repository that is needed to be scanned through valgrind, default is develop
+COLLECT_FILES=${3}
```
Comment (ashish-jabble) [inline tests/system/memory_leak/scripts/setup:6]: Always better argument in double quotes "$3"

**Review by ashish-jabble — APPROVED** [review]
Not sure If [FOGL-7839](https://github.com/fledge-iot/fledge/compare/FOGL-7839) branch is required as Mark has committed some of the commits onto same branch and rebased with this.
Either we need to change its base branch to FOGL-7839 or get it first merged into develop and then Rebase FOGL-7839 branch with develop

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**